ITE/6/README.md

```diff
@@ -0,0 +1,929 @@
+# ITE-6: Generalized link format
+
+Author: lodato@google.com
+
+Contributors/reviewers:
+  dedic@google.com, nitinjain@google.com, patricklawson@google.com,
+  tomhennen@google.com, wietse@google.com
+
+Date: September 2020
+
+# Abstract
+
+This ITE defines a new schema for in-toto link files, which are now generally
+called “attestations.” An attestation has three distinct layers, mapping to the
+three distinct steps in verification. The innermost layer is user-defined to
+allow customers to define their own schemas; “link” is now one such user-defined
+schema.
+
+This specification is developed jointly with
+[Binary Authorization](https://cloud.google.com/binary-authorization), who will
+support the agreed-upon format once finalized. This way a single link file will
+be usable by either system.
+
+# Specification
+
+## Introduction
+
+An **attestation** is the generalization of an in-toto link. It is a statement
+about an artifact, signed by an attester. Each attestation has a type indicating
+what the statement means, plus type-dependent details. In this new model, a
+“link” is just one type of attestation, but others are possible.
+
+Examples of attestations:
+
+*   Provenance: GitHub Actions attests to the fact that it built a container
+    image with digest “sha256:87f7fe…” from git commit “f0c93d…” in the “master”
+    branch of “https://github.com/example/foo”.
+*   Code review: GitHub attests to the fact that Alice uploaded and Bob approved
+    git commit “f0c93d…” in the “master” branch of
+    “https://github.com/example/foo”.
+*   Test result: Google Container Analysis attests to the fact that no
+    vulnerabilities were found in container image “sha256:87f7fe…” at a
+    particular time.
+*   Policy decision: Binary Authorization attests to the fact that container
+    image “sha256:87f7fe…” is allowed to run under GKE project “example-project”
+    within the next 4 hours, and that it used the three attestations above and
+    as well as the policy with sha256 hash “79e572”.
+
+## Model and Terminology
+
+An attestation has three layers, corresponding to the three logic steps in
+verifying an attestation:
+
+*   **Authentication**: Ensures authenticity and integrity of the attestation.
+    Handles:
+    *   Who signed the attestation.
+    *   How to interpret the next layer (serialization).
+*   **Binding**: Ensures that the attestation is relevant for the verification
+    context. Handles:
+    *   What this attestation means, via a type ID.
+    *   What artifacts this attestation is about.
+*   **Details**: Additional data, specified with a type-dependent schema.
+
+The binding and details together are called the **claim**.
+
+![drawing](attestation_parts.png)
+
+## Authentication and serialization
+
+[ITE-5](https://github.com/MarkLodato/ITE/blob/ite-5/ITE/5/README.md) defines a
```

Comment:
NOTE: If accepted this will be the signing spec (https://github.com/secure-systems-lab/signing-spec/pull/2) instead of ITE-5.

Reply:
Clarified.